build/lib/SpinCore_pp/SpinCore_pp.py

```diff
@@ -162,28 +165,28 @@ def apply_cycles(ppg_in,list_of_cycles_found):
             found_cycle = True
             break
     if found_cycle:
-        print "found phase cycle named",cycle_name
+        logging.debug(strm("found phase cycle named",cycle_name))
         all_cycles = [ppg_element[3] for ppg_element in ppg_in if (((ppg_element[0] == 'pulse' or ppg_element[0] == 'pulse_TTL') and ppg_element[2] == cycle_name) if len(ppg_element)>3 else False )]
         maxlen = max(map(len,all_cycles))
         list_of_cycles_found = [(cycle_name,maxlen)] + list(list_of_cycles_found)
-        print "LIST OF CYCLES FOUND NOW"
+        logging.debug("LIST OF CYCLES FOUND NOW")
         ppg_out = []
-        for j in xrange(maxlen):
+        for j in range(maxlen):
             for k,ppg_elem in enumerate(ppg_in):
                 if len(ppg_elem) > 3 and ppg_elem[2] == cycle_name:
                     elem_copy = list(ppg_elem)
                     spec_len = len(elem_copy[3])
                     c_idx = j % spec_len
-                    ppg_out.append(tuple(elem_copy[:2]+[elem_copy[3][c_idx]]))
+                    ppg_out.append(tuple(elem_copy[:2]+[elem_copy[3][c_idx].item()]))
```

Comment:
explain in a comment what is going on here.

Reply:
we are adding the element to the ppg_out list. elem_copy[:2] would be index 0, and 1 of the elem_copy. But I was getting an error when running a script (can't remember off the top of my head what specifically but there was an issue with combining type of elem_copy[:2] with type of [elem_copy[3][c_idx]) combining list type ([elem_copy[3][c_idx]] with the elec_copy[:2] item. So in order to allow this addition I added the item()